src/pipeline/rackhd/docker/Publish.groovy

```diff
@@ -0,0 +1,50 @@
+package pipeline.rackhd.docker
+
+/*
+publish rackhd
+:library_dir: the directory of on-build-config
+:docker_tar_path: where the docker image tar lies
+:dockerhub_user: the rackhd user of dockerhub
+:dockerhub_pass: the rackhd user password of dockerhub
+*/
+def publish(String library_dir, String docker_tar_path, String dockerhub_user, String dockerhub_pass) {
+        timeout(120){
+        	sh """#!/bin/bash -ex
+        	pushd $library_dir/src/pipeline/rackhd/docker
+        	./publish.sh --DOCKER_STASH_PATH $docker_tar_path \
```

Comment:
do  you need to run publish.sh with sudo ?

Reply:
No, it works fine without sudo. :)